src/Bicep.Core/Registry/OciArtifactRegistry.cs

```diff
@@ -175,7 +175,7 @@ public static string GetPublicBicepModuleDocumentationUri(string publicModuleNam
             return $"https://github.com/Azure/bicep-registry-modules/tree/{publicModuleName}/{tag}/modules/{publicModuleName}/README.md";
         }
 
-        public override Task<string?> TryGetDescription(OciArtifactReference ociArtifactModuleReference)
+        public override Task<string?> TryGetDescription(ModuleSymbol module, OciArtifactReference ociArtifactModuleReference)
```

Comment:
nit: should `module` be renamed to `_` since its not in use?

Reply:
That's not a convention we follow for function arguments.

Reply:
Good point about the rename - I'll rename to `TryGetModuleDescription`.

My personal take is that the `IArtifactRegistry` abstraction isn't really capable of being truly agnostic to whether an artifact is a module or a provider, and we should stop trying to pretend it is 😄. Questions like "What is the description?" simply don't make sense for a provider.

There are some places where it's useful to be agnostic (such as restoration logic - you wouldn't want to have to write it twice), but that doesn't mean we should try and shoehorn in agnosticism in places where it doesn't make sense.